rs/ic-observability/multiservice-discovery/tests/tests.rs

```diff
@@ -0,0 +1,131 @@
+// Note: If you need to update the assertions, please check the README.md file for guidelines.
+#[cfg(test)]
+mod tests {
+    use assert_cmd::cargo::CommandCargoExt;
+    use multiservice_discovery_shared::builders::prometheus_config_structure::{
+        PrometheusStaticConfig, IC_NAME, IC_NODE, IC_SUBNET, JOB
+    };
+    use serde_json::Value;
+    use tempfile::tempdir;
+    use std::collections::{BTreeMap, BTreeSet};
+    use std::process::Command;
+    use std::time::Duration;
+    use std::thread;
+    use tokio::runtime::Runtime;
+
+    const BAZEL_SD_BIN: &str = "rs/ic-observability/multiservice-discovery/multiservice-discovery";
+    const CARGO_SD_BIN: &str = "multiservice-discovery";
+
+    async fn fetch_targets() -> anyhow::Result<Vec<PrometheusStaticConfig>> {
+        const TARGETS_URL: &str = "http://localhost:8000/prom/targets";
+        let timeout_duration = Duration::from_secs(500);
+        let start_time = std::time::Instant::now();
+
+        loop {
+            if start_time.elapsed() > timeout_duration {
+                return Err(anyhow::anyhow!("Timeout reached"));
+            }
+            if let Ok(response) = reqwest::get(TARGETS_URL).await {
+                if response.status().is_success() {
+                    let text = response.text().await?;
+                    let targets: Vec<PrometheusStaticConfig> = serde_json::from_str(&text).unwrap();
+                    return Ok(targets);
+                }
+            }
+            thread::sleep(Duration::from_secs(5));
+        }
+    }
+
+
+    async fn expected_nodes_and_subnets() -> Result<(Vec<String>, Vec<String>), reqwest::Error> {
+        const API_NODES_URL: &str = "https://ic-api.internetcomputer.org/api/v3/nodes";
```

Comment:
This is a very cool idea. Nice!
However, we should also think of a way to handle non-working public dashboard API, since we now introduce hard dependence on them being online at all times for our tests to work. And they don't have 100% uptime. And if they are down, we may not be able to (quickly) merge a PR that's needed to bring them back up.

Reply:
The only other thing I see is to spin up a whole network in the CI and use that. Generally we will never be able to have 100% non-flaky test

Reply:
I was thinking more of making the process of downloading the contents more robust. I'll add more comments below.

Reply:
@sasa-tomic I want to add support for create the `expected_data`  by running the SD from the main bin. however we need a way to publishing the artifact public.